### ISSUE TITLE
CIDEr metrics : did you always used CIDEr-D ?

### ISSUE THREAD

**Issue by gautierevn** [yaolinli/IDC#4]
First, thanks for this repo, it is a great ressource and a very interesting work.

I wondered, in the paper you mention both CIDEr and CIDEr-D as main metrics. Your code only use CIDEr-D implementation for CLEVR-Change Dataset while you refer to CIDEr (just CIDEr) in Table 3. Is it intended ? I found it disturbing.

pycocoevalcap.cider.cider use the ciderD implementation since some times it seems.

TLDR : did you always used CIDEr-D and mentioned it as CIDEr ?

Thanks for your time @yaolinli

**Comment by yaolinli** [comment]
Thanks for your interest. We use CIDEr for the CLEVR-Change dataset (Table 2 in the paper) and CIDEr-D (Table 1 in the paper) for the Birds-to-Words dataset following previous works. The eval code will give both CIDEr and CIDEr-D metrics, and you can choose the one you need.

**Comment by gautierevn** [comment]
Thanks for your answer. 

I am just confused by the fact that the scripts for CIDEr and CIDEr-D are the same. 

IDC/clver/cider_scorer.py is the same as IDC/eval/ciderD/ciderD_scorer.py, besides a "df_mode" which can leads to different results when set to "coco-val-df"; and it is not the default case. 
So the computation of these two metrics are actually rather the same.

I must be missing something, what do you think ?

**Comment by yaolinli** [comment]
Thanks for your reminder. We use CIDEr-D eval code from [https://github.com/vrama91/cider](https://github.com/vrama91/cider)  and default CIDEr eval code from [pycocoevalcap](https://github.com/salaniz/pycocoevalcap/tree/master). I have not compared the two versions carefully before. After your reminder, I found that the cider_scorer.py is exactly the same as ciderD_scorer.py because the pycocoevalcap tools use CIDEr-D implementation to calculate CIDEr by default. If you want to calculate two different scores (CIDEr and CIDEr-D), you can refer to the CIDEr eval code in the  [https://github.com/vrama91/cider](https://github.com/vrama91/cider)  repo. However, I think the pycocoevalcap is a standard tool to evaluate visual (image/video) captioning tasks nowadays. If we use the original CIDEr code from vrama91, it may be unfair to compare the results with other works. In my opinion, we can keep using the default CIDEr from pycocoevalcap :)

**Comment by gautierevn** [comment]
Thanks a lot for your answer, it was very clear and helped me to understand how this metric was used in this field :)
